kaiaai_telemetry_node:
  ros__parameters:

    laser_scan:
      mask_radius_meters: 0.1
      topic_name_pub: "scan"
      frame_id: "base_scan"
      discard_broken_scans: true
#      lidar_model: "LDS02RR"

    lidar:
      model:            ['YDLIDAR-X4','XIAOMI-LDS02RR','YDLIDAR-X2-X2L','YDLIDAR-X3-PRO','YDLIDAR-X3-PRO','YDLIDAR-X3','NEATO-XV11','RPLIDAR-A1','DELTA-2G','DELTA-2A','DELTA-2B','LDROBOT-LD14P','CAMSENSE-X1','YDLIDAR-SCL']
      angle_offset_deg: [         0.0,          -180.0,             0.0,             0.0,             0.0,         0.0,      -180.0,      -180.0,     -90.0,     -90.0,     -90.0,            0.0,         16.0,          0.0]
      pub_scan_size:    [         720,             360,             720,             720,             720,         720,         360,         720,       360,       720,       360,            720,          360,          720]
      clockwise:        [        true,            true,            true,            true,            true,        true,       false,        true,      true,      true,      true,           true,         true,         true]
      range_min_meters: [        0.15,            0.15,             0.1,            0.12,            0.12,        0.12,        0.15,        0.15,      0.15,      0.15,       0.2,            0.1,          0.1,         0.12]
      range_max_meters: [        12.0,             6.0,             8.0,             8.0,             8.0,         8.0,         6.0,        12.0,       5.0,       8.0,       8.0,            8.0,          8.0,          8.0]
      intensity:        [       false,           false,           false,           false,           false,       false,       false,       false,     false,     false,     false,          false,        false,         true]
      # 'HLS-LFCD2', 'HLS-LFCD3', 'XIAOMI-LDS01RR', 'LDROBOT-LD20'
      #         0.0,         0.0,              0.0,
      #         360,         360,              360,
      #        true,        true,             true,
      #        0.12,        0.16,             0.15,
      #         3.5,         8.0,              9.0,

    telemetry:
      topic_name_sub: "telemetry"

    tf:
      frame_id: "odom"
      child_frame_id: "base_footprint"

    joints:
      topic_name_pub: "joint_states"
      wheel:
        right: "wheel_right_joint"
        left: "wheel_left_joint"

    odometry:
      frame_id: "odom"
      child_frame_id: "base_footprint"
      topic_name_pub: "odom"

    battery:
      topic_name_pub: "battery_state"
      voltage_full: 25.2
      voltage_empty: 22.2

    wifi:
      topic_name_pub: "wifi_state"
